scripts/core/exp-bindless-images.yml

```diff
@@ -0,0 +1,473 @@
+type: header
+desc: "Bindless Images Extension APIs"
+ordinal: "5"
+--- #--------------------------------------------------------------------------
+type: enum
+desc: "Dictates the type of memory copy."
+class: $xBindlessImages
+name: $x_exp_image_copy_flags_t
+etors:
+  - name: HTOD
+    desc: "Host to device."
+  - name: DTOH
+    desc: "Device to host"
+  - name: DTOD
+    desc: "Device to device"
+--- #--------------------------------------------------------------------------
+type: enum
+desc: "Dictates the method in which sampling between mipmap levels is performed."
+class: $xBindlessImages
+name: $x_exp_sampler_mip_filter_mode_flags_t
+etors:
+  - name: NEAREST
+    desc: "Filter mode nearest."
+  - name: LINEAR
+    desc: "Filter mode linear."
+--- #--------------------------------------------------------------------------
+type: function
+desc: "Destroy bindless unsampled image handles"
+class: $xBindlessImages
+name: UnsampledImageHandleDestroyExp
+ordinal: "0"
+analogue:
+    - "**cuSurfObjectDestroy**"
+params:
+    - type: $x_context_handle_t
+      name: hContext
+      desc: "[in] handle of the context object"
+    - type: $x_exp_image_handle_t
+      name: hImage
+      desc: "[in] pointer to handle of image object to destroy"
+returns:
+    - $X_RESULT_ERROR_INVALID_CONTEXT
+    - $X_RESULT_ERROR_INVALID_VALUE
+    - $X_RESULT_ERROR_INVALID_MEM_OBJECT
+--- #--------------------------------------------------------------------------
+type: function
+desc: "Destroy bindless sampled image handles"
+class: $xBindlessImages
+name: SampledImageHandleDestroyExp
+ordinal: "0"
+analogue:
+    - "**cuTexObjectDestroy**"
+params:
+    - type: $x_context_handle_t
+      name: hContext
+      desc: "[in] handle of the context object"
+    - type: $x_exp_image_handle_t
+      name: hImage
+      desc: "[in] pointer to handle of image object to destroy"
+returns:
+    - $X_RESULT_ERROR_INVALID_CONTEXT
+    - $X_RESULT_ERROR_INVALID_VALUE
+    - $X_RESULT_ERROR_INVALID_MEM_OBJECT
+--- #--------------------------------------------------------------------------
+type: function
+desc: "Allocate memory for bindless images"
+class: $xBindlessImages
+name: ImageAllocateExp
+ordinal: "0"
+analogue:
+    - "**cuArray3DCreate**"
+    - "**cuMipmappedArrayCreate**"
+params:
+    - type: $x_context_handle_t
+      name: hContext
+      desc: "[in] handle of the context object"
+    - type: "const $x_image_format_t*"
+      name: pImageFormat
+      desc: "[in] pointer to image format specification"
+    - type: "const $x_image_desc_t*"
+      name: pImageDesc
+      desc: "[in] pointer to image description"
+    - type: $x_exp_image_mem_handle_t*
+      name: phImageMem
+      desc: "[out] pointer to handle of image memory allocated"
+returns:
+    - $X_RESULT_ERROR_INVALID_CONTEXT
+    - $X_RESULT_ERROR_INVALID_VALUE
+    - $X_RESULT_ERROR_INVALID_IMAGE_FORMAT_DESCRIPTOR:
+      - "`pImageDesc && UR_MEM_TYPE_IMAGE1D_BUFFER < pImageDesc->type`"
```

Comment:
I wonder if this is needed - I think the api generator will generate this check anyway.

Reply:
No you're right in this case - ignore all my previous comments 👍